Galaxus.Functional/(Option)/(Features)/Option.Match.cs

```diff
@@ -35,6 +36,37 @@ public void Match(Action<T> onSome, Action onNone)
             }
         }
 
+        /// <summary>
+        ///     Provides access to <b>self</b>'s content by calling <paramref name="onSomeAsync" /> and passing in <b>Some</b>
+        ///     or calling <paramref name="onNoneAsync" />.
+        /// </summary>
+        /// <param name="onSomeAsync">
+        ///     Called when <b>self</b> contains <b>Some</b>. The argument to this action is never the <b>null</b>
+        ///     reference.
+        /// </param>
+        /// <param name="onNoneAsync">Called when <b>self</b> contains <b>None</b>.</param>
+        public async Task MatchAsync(Func<T, Task> onSomeAsync, Func<Task> onNoneAsync)
```

Comment:
Once you've moved the code into an extension method, we should handle all the possibilities. Match is really a pain though, 'cause there are many possibilities.

For non-task options:
* `public static async Task MatchAsync<T>(this Option<T>, Func<T, Task> onSome, Func<Task> onNone);`
* `public static async Task MatchAsync<T>(this Option<T>, Action<T> onSome, Func<Task> onNone);`
* `public static async Task MatchAsync<T>(this Option<T>, Func<T, Task> onSome, Action onNone);`

For task-options:
* `public static async Task MatchAsync<T>(this Task<Option<T>>, Func<T, Task> onSome, Func<Task> onNone);`
* `public static async Task MatchAsync<T>(this Task<Option<T>>, Action<T> onSome, Func<Task> onNone);`
* `public static async Task MatchAsync<T>(this Task<Option<T>>, Func<T, Task> onSome, Action onNone);`
* `public static async Task MatchAsync<T>(this Task<Option<T>>, Action<T> onSome, Action onNone);`

And all the same again for match-to-value, so
* `public static async Task<U> MatchAsync<T, U>(this Option<T>, Func<T, Task<U>> onSome, Func<Task<U>> onNone);`
* ... and all the others again

Good thing is, though: With the two you've implemented already, you can handle all the others as well. You can take inspiration from `AsyncResultExtensions` for that ;)

Reply:
Thanks a lot! I think the latest version includes all the necessary implementations